### ISSUE TITLE
Bug: After OpenLibrary & Archive.org waitlist book is available it wont download/ open via ADE?

### ISSUE THREAD

**Issue by xbliss** [internetarchive/openlibrary#963]
https://ebooks.stackexchange.com/questions/7801/after-openlibrary-archive-org-waitlist-book-is-available-it-wont-download-ope 

After OpenLibrary & Archive.org waitlist book is available it wont download/ open via ADE? 

- Got in waitlist for: 
https://openlibrary.org/books/OL697528M/The_father_of_spin 

- Its now available to me and sends me from OL site to: 
https://archive.org/stream/fatherofspinedwa00tyel#page/n5/mode/2up 

- It shows the `Online book reader` and it shows 2 major buttons;  
>Return Book **|** PDF/ePub download 

 - Clicking on the Download button pops up dialog box: 

<BLOCKQUOTE>Select Download Format  
Your loan will last two weeks from the initial borrow date.

- Encrypted Adobe PDF
High Quality Page ImagesEncrypted 

- Adobe ePub
Smaller File, May Contain Errors

- Install Adobe Digital Editions

`In order to access your downloaded book you will need Adobe-compliant software on your device. The Internet Archive will administer this loan, but Adobe may also collect some information.` 
</BLOCKQUOTE>

I already have a workind ADE for long time where have gotten other ePubs and PDFs from OL before and then returned it (maybe not from archive.org) and some issue here.  

- Whether I choose PDF or ePUB, this Archive.org source fires of this 

  - URL: `https://lending6.us.archive.org:8080/fulfillment/URLLink.acsm?action=enterloan&ordersource=builtin&orderid=1d7fa007-1464-4f6c-97e7-902d74b71111&resid=urn%3Auuid%3Af0dad698-6c2e-4919-a80f-61c6c289b00f&gbauthdate=05%2F18%2F2018+13%3A04&dateval=1526648654&gblver=4&rights=%24lat%231527858205%24&auth=20405a0343beff078187b4ae65063ef553f85033`

  - Error: 
<BLOCKQUOTE>This site can’t provide a secure connection  

**lending6.us.archive.org** sent an invalid response.   

ERR_SSL_PROTOCOL_ERROR</BLOCKQUOTE>

**Comment by mekarpeles** [comment]
Thanks @xbliss! I'm expediting this issue and will coordinate directly with the engineers who manage our lending infrastructure. Will hopefully update this ticket soon.

cc: @rchrd2

**Comment by mekarpeles** [comment]
My suspicion is that what's happening is someone borrows the book (for 2 weeks) and then they download the epub or pdf (causing an ACS Adobe Digital Editions loan to occur). The reader then returns their book, but there's still time left on the ACS loan and so it blocks/prevents others from accessing it. I'm not sure what the solution will be (as some of these limits are a consequence of the technology setup as opposed to our preferences). Again, hopefully we'll have an update soon (thanks in advance for your patience)

**Comment by redranamber** [comment]
I have had this happen too. By forcing it to use http instead of https I got it to download.

**Comment by xbliss** [comment]
I will disable the HTTPS plugin and try again. 

I am back on Waitlist so will report back once it's available. 

PS: Not sure if this is connected but was able to get the PDF but not thr ePub for this book. 

https://openlibrary.org/works/OL16288250W/Public_relations_Edward_L._Bernays_and_the_American_scene

Brings up: 
https://archive.org/stream/publicrelationse00bernrich?ref=ol#page/n0/mode/2up 

> Select Download Format
> Choose a format to begin your download. 
> 
> - PDF : Works and I get it. 
> - ePub : "Sorry, epub creation failed" 
> - Plain Text: Get the Text 
> - DAISY: Get a Zip file 
> - Kindle: "Sorry, epub creation failed" 
> 

Thoughts on whether ePub Mobi could be gotten here?

**Comment by xbliss** [comment]
> I have had this happen too. By forcing it to use http instead of https I got it to download. 

Tried that with another book and same problem. 

> This site can’t provide a secure connection
> 
> lending6.us.archive.org sent an invalid response.
> ERR_SSL_PROTOCOL_ERROR

**Comment by tabshaikh** [comment]
Yes a bit follow up on this issue the server is returning status code 400 maybe some invalid value in the json

**Comment by tabshaikh** [comment]
![screenshot from 2018-06-05 21-44-47](https://user-images.githubusercontent.com/31198893/40988956-16112c5e-690a-11e8-8211-b9bdc48dc327.png)

**Comment by xbliss** [comment]
@redranamber  - Can you actually try this and tell me which way works with Steps? 

It should not take a lot for you to Prove or Disprove your hypothesis.

**Comment by xbliss** [comment]
@tabshaikh - Thanks for that update. Is there anything I can do to help? 

Should I share example of the other book? Has using another browser helped resolving it? I am curious.

**Comment by redranamber** [comment]
First you need a downloader, I am using [uGet](https://sourceforge.net/projects/urlget/).
Then when you try to download the book you will see "Secure Connection Failed" with the download url, mine looks like
https://lending6.us.archive.org:8080/fulfillment/URLLink.acsm?action=enterloan&ordersource=builtin&orderid=c32250fd-e1a4-4e18-a63d-2a631b271111&resid=urn%3Auuid%3A9291df5a-a9bc-44d0-9c8f-f0a3c21a2f90&gbauthdate=07%2F10%2F2018+06%3A50&dateval=1531205407&gblver=4&rights=%24lat%231532414987%24&auth=0f4a3829a0f38b6c06e7fb3e070c7e1393619eac

copy this and paste it into your downloader and change it from https to http. Then it downloads successfully.

**Comment by mekarpeles** [comment]
cc: @rchrd2 and @bfalling I think the emails which are sent need to be updated to use http instead of https? This seems like a fairly high priority issue.

**Comment by rchrd2** [comment]
@mekarpeles We don't send emails with links directly to ACS files. The ACS files are accessed through bookreader. At the moment, that server only supports http. In order to add https support, we need to proxy the server, either through Nginx, or through an API on petabox.

**Comment by xbliss** [comment]
@redranamber - Have you actually tried these steps out or theorizing that it may work? Cause I've already tried a lot of suggestions given above by people that were not tried out and did not work. 

I am on Windows so it will be a different downloader than uGet but I am sure between the few I have it could be made to do what you asked. 

I did try the http:// thing using Chrome Download manager but I will try with a separate one like FDM or GetRight
